library/core/src/num/int_macros.rs

```diff
@@ -56,6 +56,65 @@ macro_rules! int_impl {
         #[stable(feature = "int_bits_const", since = "1.53.0")]
         pub const BITS: u32 = <$UnsignedT>::BITS;
 
+        /// Tests the value of a specific bit.
+        ///
+        /// # Examples
+        ///
+        /// ```
+        /// #![feature(get_set_bit)]
+        ///
+        #[doc = concat!("let n = 0b1000001", stringify!($SelfT), ";")]
+        ///
+        /// assert!(n.bit(0));
+        /// assert!(!n.bit(1));
+        /// assert!(!n.bit(2));
+        /// assert!(!n.bit(3));
+        /// assert!(!n.bit(4));
+        /// assert!(!n.bit(5));
+        /// assert!(n.bit(6));
+        /// ```
+        #[unstable(feature = "get_set_bit", issue = "147702")]
+        #[rustc_const_unstable(feature = "get_set_bit", issue = "147702")]
+        #[must_use = "this returns the result of the operation, \
+                      without modifying the original"]
+        #[inline]
+        #[track_caller]
+        pub const fn bit(self, index: u32) -> bool {
+            self & (1 as $SelfT) << index != (0 as $SelfT)
```

Comment:
According to https://github.com/rust-lang/libs-team/issues/667#issuecomment-3403074426 the API that was accepted involves checking `index` in builds that contain overflow checking (debug mode or `-C overflow-checks`). I tested this and it did not panic:

```rust
#![feature(get_set_bit)]
fn main() {
    println!("{}", 0i32.bit(40));
}
```

whereas this does panic:

```rust
#![allow(arithmetic_overflow)]
fn main() {
    let index = 40;
    println!("{}", 0i32 & (1 as i32) << index != (0 as i32));
}
```

I think this attribute does the thing:

https://github.com/rust-lang/rust/blob/5f9dd05862d2e4bceb3be1031b6c936e35671501/library/core/src/num/int_macros.rs#L3289